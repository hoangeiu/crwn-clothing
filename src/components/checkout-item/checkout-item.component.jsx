import React from "react";

import "./checkout-item.styles.scss";

const CheckoutItem = ({
  cartItem: { id, name, imageUrl, price, quantity },
}) => {
  return (
    <div className="checkout-item">
      <div className="image-container">
        <img src={imageUrl} atl="item" />
      </div>
      <span className="name">{name}</span>
      <span className="quantity">{quantity}</span>
      <span className="price">{price}</span>
      <span className="remove-button">&#10005;</span>
    </div>
  );
};

export default CheckoutItem;
