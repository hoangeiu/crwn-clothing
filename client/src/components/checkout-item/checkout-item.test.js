import React from "react";
import { shallow } from "enzyme";

import { CheckoutItem } from "./checkout-item.component";

const findTestByAttr = (wrapper, val) => wrapper.find(`[data-test="${val}"]`);

describe("CheckoutItem component", () => {
  let wrapper;
  let mockClearItem;
  let mockAddItem;
  let mockRemoveItem;
  beforeEach(() => {
    mockClearItem = jest.fn();
    mockAddItem = jest.fn();
    mockRemoveItem = jest.fn();

    const mockProps = {
      cartItem: {
        name: "Gucci hat",
        imageUrl: "www.test.com",
        price: 100,
        quantity: 5,
      },
      clearItem: mockClearItem,
      addItem: mockAddItem,
      removeItem: mockRemoveItem,
    };

    wrapper = shallow(<CheckoutItem {...mockProps} />);
  });

  it("should CheckoutItem renders without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call removeItem when button is clicked", () => {
    // wrapper.find(".quantity").childAt(0).simulate("click");
    const removeButton = findTestByAttr(wrapper, "removeItem");
    removeButton.simulate("click");

    expect(mockRemoveItem).toHaveBeenCalled();
  });

  it("should call addItem when button is clicked", () => {
    // wrapper.find(".quantity").childAt(2).simulate("click");
    const addButton = findTestByAttr(wrapper, "addItem");
    addButton.simulate("click");

    expect(mockAddItem).toHaveBeenCalled();
  });

  it("should call clearItem when button is click", () => {
    wrapper.find(".remove-button").simulate("click");

    expect(mockClearItem).toHaveBeenCalled();
  });
});
