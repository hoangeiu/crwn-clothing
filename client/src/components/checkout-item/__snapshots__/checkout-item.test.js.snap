// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component should CheckoutItem renders without error 1`] = `
<div
  className="checkout-item"
>
  <div
    className="image-container"
  >
    <img
      atl="item"
      src="www.test.com"
    />
  </div>
  <span
    className="name"
  >
    Gucci hat
  </span>
  <span
    className="quantity"
  >
    <div
      className="arrow"
      data-test="removeItem"
      onClick={[Function]}
    >
      ❮
    </div>
    <span
      className="value"
    >
      5
    </span>
    <div
      className="arrow"
      data-test="addItem"
      onClick={[Function]}
    >
      ❯
    </div>
  </span>
  <span
    className="price"
  >
    100
  </span>
  <span
    className="remove-button"
    onClick={[Function]}
  >
    ✕
  </span>
</div>
`;
